### DISCUSSION TITLE
Are there zoom or scroll callbacks?

### DISCUSSION THREAD

**Discussion by jrmoulton** [slint-ui/slint#680] [Q&A] [answered]
I can see that some views like ScrollView can be scrolled but do those views have associated callbacks for scrolling and or zooming?
Is there any way to have a callback for when a zoom happens on a touchpad for example?

**Comment by ogoffart** [comment, marked as answer]
Currently, the lower API that does scrolling is the Flickable.

We don't expose public api for zooming yet.

What use case do you have in mind?

**Comment by jrmoulton** [comment]
I'm building a document viewer for markdown files and I was trying to build in the ability to zoom by pinching.

**Comment by CRogers** [comment]
I would also like a way to zoom or easily scale a bunch of subcomponents. I'm starting out building a circuit designer (for simple stuff like AND/OR gates) and being able to zoom in/out (and likely pan) the whole set of gates and connections would be great (rather than having to manually move/scale all the components myself).
